src/ORM/UniqueKey/UniqueKeyService.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace SilverStripe\ORM\UniqueKey;
+
+use Exception;
+use SilverStripe\Core\ClassInfo;
+use SilverStripe\Core\Injector\Injectable;
+use SilverStripe\ORM\DataObject;
+
+/**
+ * Class UniqueKeyService
+ *
+ * Generate a unique key for data object
+ *
+ * recommended use:
+ * - when you need unique key for caching purposes
+ * - when you need unique id on the front end (for example JavaScript needs to target specific element)
+ *
+ * @package SilverStripe\ORM\UniqueKey
+ */
+class UniqueKeyService implements UniqueKeyInterface
+{
+    use Injectable;
+
+    /**
+     * @param DataObject $object
+     * @param array $keyComponents
+     * @return string
+     * @throws Exception
+     */
+    public function generateKey(DataObject $object, array $keyComponents = []): string
+    {
+        $id = $object->isInDB() ? (string) $object->ID : bin2hex(random_bytes(16));
+        $class = ClassInfo::shortName($object);
+        $keyComponents = json_encode($keyComponents);
```

Comment:
what happens when you pass in a DataObject to json_encode?

I do like the use of json_encode, though possible maybe just update the `@param array $keyComponents` to say pass in an array of strings since [$dataObject->SomeColumn] is probably as complex $keyComponents as it'll ever need to be